test/clj/rems/api/test_extra_pages.clj

```diff
@@ -1,29 +1,62 @@
+  (test-data/create-test-api-key!)
+  (test-data/create-test-users-and-roles!)
+  (with-redefs [rems.config/env (assoc rems.config/env
+                                       :extra-pages [{:id "test"
+                                                      :filename "test-en.md"
+                                                      :translations {:fi {:title "Testi"
+                                                                          :filename "test-fi.md"}
+                                                                     :en {:title "Test"}
+                                                                     :de {:url "https://example.org/de"}}}
+
+                                                     {:id "test2"
+                                                      :url "https://example.org/"
+                                                      :translations {:fi {:title "Testi"
+                                                                          :filename "test-fi.md"}
+                                                                     :en {:title "Test"
+                                                                          :filename "test-en.md"}
+                                                                     :de {:url "https://example.org/de"}}}
+
+                                                     {:id "test-roles"
+                                                      :roles [:handler]
+                                                      :filename "test-en.md"}]
+                                       :extra-pages-path "./test-data/extra-pages/")]
+    (testing "fetch existing extra-page"
+      (let [response (api-call :get "/api/extra-pages/test" {} "42" "alice")]
+        (is (= (set (keys response)) #{:en :fi :de}) "should return content in defined languages")
+        (is (= "This is a test.\n" (:en response)) "should have fallback")
+        (is (= "Tämä on testi.\n" (:fi response)) "should have custom localization")
+        (is (= "This is a test.\n" (:de response)) "should have fallback though url is intended for front-end"))
+
+      (let [response (api-call :get "/api/extra-pages/test2" {} "42" "alice")]
+        (is (= (set (keys response)) #{:en :fi :de}))
+        (testing "without fallback"
+          (is (nil? (:de response)) "should have no content as url is intended for front-end"))))
+
+    (testing "roles"
+      (testing "without the role"
+        (let [response (-> (request :get "/api/extra-pages/test-roles")
+                           (authenticate "42" "alice")
                            handler)]
-          (is (response-is-not-found? response)))))))
+          (is (response-is-not-found? response))))
+
+      (testing "with the role"
+        (let [response (-> (request :get "/api/extra-pages/test-roles")
+                           (authenticate "42" "handler")
+                           handler)]
+          (is (response-is-not-found? response)))))
```

Comment:
should the response be contents of the `test-en.md` file here instead of `not-found`?

Reply:
Good catch. There was a small bug in the implementation because of this (not looping configured languages, but only defined languages, where there weren't any here)